File: app/src/main/java/com/apkide/core/LoggerReceiver.java
package com.apkide.core;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

public class LoggerReceiver extends BroadcastReceiver {
	private static final String TAG = "LoggerReceiver";
	public final static String ACTION_LOGGER = "ide.intent.action.logger";
	
	@Override
	public void onReceive(Context context, Intent intent) {
		if (!intent.getAction().equals(ACTION_LOGGER))return;
		Logger.LogRecord record = intent.getParcelableExtra("record");
		Log.d(TAG, record.getLoggerName() + ":" + record.getLevel().name + ":" + record.getMessage());
	}
}

File: app/src/main/java/com/apkide/core/logging/LoggerReceiver.kt
package com.apkide.core.logging

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class LoggerReceiver : BroadcastReceiver() {
	private var _loggerListener: LoggerListener? = null
	override fun onReceive(context: Context, intent: Intent) {
		val record = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
			intent.getParcelableExtra("record", LogRecord::class.java)
		} else {
			intent.getParcelableExtra("record")
		}
		record ?: return
		if (DEBUG)
			Log.d(TAG, record.loggerName + ":" + record.level.toString() + ":" + record.message)
		
		if (_loggerListener != null) _loggerListener!!.onLogging(
			record.loggerName,
			record.level,
			record.loggerName,
			record.throwable
		)
	}
	
	fun setLoggerListener(loggerListener: LoggerListener?) {
		_loggerListener = loggerListener
	}
	
	interface LoggerListener {
		fun onLogging(logName: String?, level: Level?, msg: String?, thrown: Throwable?)
	}
	
	companion object {
		private const val TAG = "LoggerReceiver"
		private const val DEBUG = true
		const val ACTION_LOGGER = "ide.intent.action.logger"
	}
}